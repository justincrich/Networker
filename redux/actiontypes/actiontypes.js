//ContactList Actions
export const CONTACTLIST_GET_ALL = 'CONTACTLIST_REQUEST_GET_ALL';
export const CONTACTLIST_BEGIN_GET_ALL = 'CONTACTLIST_BEGIN_GET_ALL';
export const CONTACTLIST_RECEIVE_ALL = 'CONTACTLIST_RECEIVE_GET_ALL';

export const CONTACT_REQUEST_ONE = 'CONTACT_REQUEST_ONE';

//Contact Create
export const CONTACT_REQUEST_CREATE = 'CONTACT_REQUEST_CREATE';
export const CONTACT_BEGIN_CREATE = 'CONTACT_BEGIN_CREATE';
export const CONTACT_RESOLVE_CREATE = 'CONTACT_RESOLVE_CREATE';
export const CONTACT_RESET_STATUS = 'CONTACT_RESET_STATUS';
//Contact Update
export const CONTACT_REQUEST_UPDATE = 'CONTACT_REQUEST_UPDATE';
export const CONTACT_BEGIN_UPDATE = 'CONTACT_BEGIN_UPDATE';
export const CONTACT_RESOLVE_UPDATE = 'CONTACT_RESOLVE_UPDATE';

export const CONTACT_REQUEST_DELETE = 'CONTACT_REQUEST_DELETE';
export const CONTACT_THROW_ERROR = 'CONTACT_THROW_ERROR';
// export const CONTACT_CLEAR_ERROR = 'CONTACT_CLEAR_ERROR';


