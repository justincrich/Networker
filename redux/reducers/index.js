// import { combineReducers } from 'redux';
// import ContactsDataReducer from './contacts_reducer';

// export default combineReducers({
//     contacts:ContactsDataReducer
// });