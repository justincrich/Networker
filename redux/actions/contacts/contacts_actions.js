import * as ActionTypes from '../../actiontypes/actiontypes';
import realm from '../../../data/realm';
import {ContactSchema} from '../../../data/schemas';
// import {ContactSchema} from '../../data/schemas';
//const Realm = require('realm');
// const schema = {
//     name:'Contact',
//     properties:{
//         id:{type:'int',default:Date.now()},
//         firstName:'string',
//         lastName:'string',
//         pictureUri:{type:'string',default:''},
//         email:'string?',
//         phoneNumber:'string?',
//         notes:'string?',
//         jobTitle:'string?',
//         company:'string?',
//     }
// }
//Get Contact


export function getContact(id){
    return function (dispatch){
        dispatch(beginGetContact(id));
        try{
            //let query = `id = ${id}`;
            //let contact = realm.objects('Contact').filtered(query);
            const contact = realm.objectForPrimaryKey(ContactSchema,id);
            //console.log('IN ACTION',Object.assign({},contact))
            dispatch(resolveGetContact(Object.assign({},contact)));
        }catch(e){
            dispatch(throwContactError(e))
        }
    }
}

export function beginGetContact(id){
    return{
        type: ActionTypes.CONTACT_REQUEST_ONE_BEGIN,
        id:id
    }
}
export function resolveGetContact(contact){
    return{
        type: ActionTypes.CONTACT_REQUEST_ONE_RESOLVE,
        contact:contact
    }
}

//Contact Create
export function requestContactCreate(contact){
    return function (dispatch){
        dispatch(beginContactCreate());
        try{
            
            realm.write(()=>{
                let myContact = realm.create('Contact',{
                   firstName:contact.firstName,
                   lastName:contact.lastName,
                   pictureUri: contact.pictureUri,
                   email:contact.email,
                   phoneNumber:contact.phoneNumber,
                   notes:contact.notes,
                   jobTitle:contact.jobTitle,
                   company:contact.company
               });
            });
            dispatch(resolveContactCreate())
            dispatch(getAllContacts());
        }catch(e){
            dispatch(throwContactError(e))
        }
    }
}

export function requestDeleteContact(id){
    return function (dispatch){
        dispatch(beginContactDelete(id))
        try{
            
            realm.write(()=>{
                let query = `id = ${id}`;
                let contact = realm.objects('Contact').filtered(query)[0];
                //let contact = realm.objectForPrimaryKey(ContactSchema,id);
                console.log('new contact delete!!!',contact.id)
                realm.delete(contact);
                dispatch(getAllContacts());
                dispatch(resolveContactDelete());
                dispatch(clearSelectedContact());
                
                
            })
        }catch(e){
            dispatch(throwContactError(e));
        }
        
        
    }
}

export function requestUpdateContact(id,updates){
    return function (dispatch){
        dispatch(beginContactUpdate());
        try{
            realm.write(()=>{
                let contact = realm.create('Contact',{id:id,...updates},true);
                dispatch(resolveContactUpdate(Object.assign({},contact)));
            });
        }catch(e){
            dispatch(throwContactError(e));
        }
    }
}

//Contact List

export function getAllContacts(){
    return function (dispatch){
        dispatch(beginGetAllContacts());
        try{
            let contacts = realm.objects('Contact').sorted('lastName','firstName');
            let contactArr = contacts.map(contact=>{
                return Object.assign({},contact);
            });
            dispatch(receiveAllContacts(contactArr));
            
        }catch(e){
            dispatch(throwContactError(e))
        }
        
    }
}

export function resetContactStatus(){
    return{
        type:ActionTypes.CONTACT_RESET_STATUS,
    }
}

export function clearSelectedContact(){
    return{
        type:ActionTypes.CONTACT_CLEAR_SELECTION
    }
}

/*Status Changing Methods/PRIVATE*/

//CONTACT LIST
export function beginGetAllContacts(){
    return{
        type:ActionTypes.CONTACTLIST_BEGIN_GET_ALL
    }
}
export function receiveAllContacts(contacts){
    return{
        type: ActionTypes.CONTACTLIST_RECEIVE_ALL,
        contacts:contacts
    }
}

//CONTACT CREATE
export function beginContactCreate(){
    return{
        type:ActionTypes.CONTACT_BEGIN_CREATE,
    }
}

export function resolveContactCreate(){
    return{
        type: ActionTypes.CONTACT_RESOLVE_CREATE,
    }
}

//UPDATE CONTACT

export function beginContactUpdate(){
    return{
        type:ActionTypes.CONTACT_REQUEST_UPDATE
    }
}

export function resolveContactUpdate(contact){
    return{
        type:ActionTypes.CONTACT_RESOLVE_UPDATE,
        updatedContact:contact
    }
}

//DELETE CONTACT

export function beginContactDelete(id){
    return{
        type:ActionTypes.CONTACT_REQUEST_DELETE,
        id:id
    }
}

export function resolveContactDelete(){
    return{
        type:ActionTypes.CONTACT_RESOLVE_DELETE
    }
}


export function throwContactError(e){
    return{
        type:ActionTypes.CONTACT_THROW_ERROR,
        error:e,
    }
}