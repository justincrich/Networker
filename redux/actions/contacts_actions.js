import * as ActionTypes from '../actiontypes/actiontypes';
import realm from '../../data/realm';
import {ContactSchema} from '../../data/schemas';
// import {ContactSchema} from '../../data/schemas';
//const Realm = require('realm');
// const schema = {
//     name:'Contact',
//     properties:{
//         id:{type:'int',default:Date.now()},
//         firstName:'string',
//         lastName:'string',
//         pictureUri:{type:'string',default:''},
//         email:'string?',
//         phoneNumber:'string?',
//         notes:'string?',
//         jobTitle:'string?',
//         company:'string?',
//     }
// }
//Get Contact


export function getContact(id){
    return function (dispatch){
        dispatch(beginGetContact(id));
        try{
            const contact = realm.objectForPrimaryKey(ContactSchema,id);
            console.log('IN ACTION',contact)
            dispatch(resolveGetContact(contact));
        }catch(e){
            dispatch(throwContactError(e))
        }
    }
}

function beginGetContact(id){
    return{
        type: ActionTypes.CONTACT_REQUEST_ONE_BEGIN,
        id:id
    }
}
function resolveGetContact(contact){
    return{
        type: ActionTypes.CONTACT_REQUEST_ONE_RESOLVE,
        contact:contact
    }
}

//Contact Create
export function requestContactCreate(contact){
    return function (dispatch){

        dispatch(beginContactCreate());
        try{
            
            realm.write(()=>{
                let myContact = realm.create('Contact',{
                   firstName:contact.firstName,
                   lastName:contact.lastName,
                   pictureUri: contact.pictureUri,
                   email:contact.email,
                   phoneNumber:contact.phoneNumber,
                   notes:contact.notes,
                   jobTitle:contact.jobTitle,
                   company:contact.company
               });
            });
            dispatch(resolveContactCreate())
            dispatch(getAllContacts());
        }catch(e){
            dispatch(throwContactError(e))
        }
    }
}

//Contact List

export function getAllContacts(){
    return function (dispatch){
        dispatch(beginGetAllContacts());
        try{
            // Realm.open({schema:[schema]})
            // .then(realm=>{
            //     let contacts = realm.objects('Contact').sorted('lastName','firstName');
            //     console.log('BLLLAAA',contacts,contacts.length);
            // })
            let contacts = realm.objects('Contact').sorted('lastName','firstName');
            
            dispatch(receiveAllContacts(Array.from(contacts.values())));
            
        }catch(e){
            dispatch(throwContactError(e))
        }
        
    }
}

export function resetContactStatus(){
    return{
        type:ActionTypes.CONTACT_RESET_STATUS,
    }
}

/*Status Changing Methods/PRIVATE*/

//CONTACT LIST
function beginGetAllContacts(){
    return{
        type:ActionTypes.CONTACTLIST_BEGIN_GET_ALL
    }
}
function receiveAllContacts(contacts){
    return{
        type: ActionTypes.CONTACTLIST_RECEIVE_ALL,
        contacts:contacts
    }
}

//CONTACT CREATE
function beginContactCreate(){
    return{
        type:ActionTypes.CONTACT_BEGIN_CREATE,
    }
}

function resolveContactCreate(){
    return{
        type: ActionTypes.CONTACT_RESOLVE_CREATE,
    }
}

function throwContactError(e){
    return{
        type:ActionTypes.CONTACT_THROW_ERROR,
        error:e,
    }
}