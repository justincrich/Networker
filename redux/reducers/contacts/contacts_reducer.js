import * as ActionTypes from '../../actiontypes/actiontypes';
import {contactsInitialState} from '../initial_states';



export default function ContactsReducer(state=contactsInitialState,action){
    switch(action.type){
        case ActionTypes.CONTACT_REQUEST_ONE_BEGIN:{
            return{
                ...state,
                requestedId:action.id,
                status:'loading'
            }
        }
        case ActionTypes.CONTACT_REQUEST_ONE_RESOLVE:{
            return{
                ...state,
                status:'ready',
                requestedId:null,
                selectedContact:action.contact
            }
        }
        case ActionTypes.CONTACTLIST_BEGIN_GET_ALL:{
            return{
                ...state,
                status:'loading'
            }
        }
        case ActionTypes.CONTACTLIST_RECEIVE_ALL:{
            return{
                ...state,
                status:'complete',
                contacts:action.contacts
            }
        }
        case ActionTypes.CONTACT_REQUEST:{
            let selection = state.contacts[action.id];
            return {
                ...state,
                selectedContact:selection
            }
        }
        case ActionTypes.CONTACT_BEGIN_CREATE:{
            return {
                ...state,
                status:'loading'
            }
        }
        case ActionTypes.CONTACT_RESOLVE_CREATE:{
            return {
                ...state,
                status:'complete'
            }
        }
        case ActionTypes.CONTACT_REQUEST_UPDATE:{
            return{
                ...state,
                status:'loading'
            }
        }
        case ActionTypes.CONTACT_RESOLVE_UPDATE:{
            return {
                ...state,
                status:'complete',
                selectedContact:action.updatedContact
            }
        }
        case ActionTypes.CONTACT_REQUEST_DELETE:{
            return{
                ...state,
                status:'loading',
                requestDeleteId:action.id
            }
        }
        case ActionTypes.CONTACT_RESOLVE_DELETE:{
            return{
                ...state,
                status:'complete'
            }
        }
        case ActionTypes.CONTACT_THROW_ERROR:{
            return {
                ...state,
                error:action.error,
                status:'error'
            }
        }
        case ActionTypes.CONTACT_CLEAR_SELECTION:{
            return{
                ...state,
                selectedContact:null
            }
        }
        case ActionTypes.CONTACT_RESET_STATUS:{
            return{
                ...state,
                status:'ready'
            }
        }
        default:
            return state;
    }
}