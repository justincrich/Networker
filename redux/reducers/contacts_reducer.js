import * as ActionTypes from '../actiontypes/actiontypes';
// import {REHYDRATE} from 'redux-persist/constants';

const initialState={

    contacts:[],
    requestedId:null,
    selectedContact:{},
    status:'ready',
    error:null,
}

export default function ContactsReducer(state=initialState,action){
    switch(action.type){
        case ActionTypes.CONTACT_REQUEST_ONE_BEGIN:{
            return{
                ...state,
                requestedId:action.id,
                status:'loading'
            }
        }
        case ActionTypes.CONTACT_REQUEST_ONE_RESOLVE:{
            return{
                ...state,
                status:'ready',
                requestedId:null,
                selectedContact:action.contact
            }
        }
        case ActionTypes.CONTACTLIST_BEGIN_GET_ALL:{
            return{
                ...state,
                status:'loading'
            }
        }
        case ActionTypes.CONTACTLIST_RECEIVE_ALL:{
            return{
                ...state,
                status:'complete',
                contacts:action.contacts
            }
        }
        case ActionTypes.CONTACT_REQUEST:{
            let selection = state.contacts[action.id];
            return {
                ...state,
                selectedContact:selection
            }
        }
        case ActionTypes.CONTACT_BEGIN_CREATE:{
            return {
                ...state,
                status:'loading'
            }
        }
        case ActionTypes.CONTACT_RESOLVE_CREATE:{
            return {
                ...state,
                status:'complete'
            }
        }
        case ActionTypes.CONTACT_THROW_ERROR:{
            return {
                ...state,
                error:action.error,
                status:'error'
            }
        }
        case ActionTypes.CONTACT_RESET_STATUS:{
            return{
                ...state,
                status:'ready'
            }
        }
        default:
            return state;
    }
}