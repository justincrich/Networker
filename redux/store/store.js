import ContactsReducer from '../reducers/contacts_reducer';
import { compose, createStore, applyMiddleware } from 'redux';
const thunk = require('redux-thunk').default; //allows you to write action creators that return a function instead of an action
import {createLogger} from 'redux-logger'; //logs out state in console



let debuggWrapper = (data) => data;
const logger = createLogger();
export const store = createStore(
    ContactsReducer,
    compose(
        applyMiddleware(thunk,logger),
        window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    )
);

// export const store = createStore(
//     ContactsReducer,
//     composeWithDevTools(applyMiddleware(thunk,logger))
    
// );
// if(__DEV__){
//     store = createStore(
//         ContactsReducer,
//         composeWithDevTools(applyMiddleware(thunk,logger))
        
//     );
// }else{
//     store = createStore(
//         ContactsReducer,
//         applyMiddleware(thunk,logger)
        
//     );
// }


