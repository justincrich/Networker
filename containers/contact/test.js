import 'react-native';
import React from 'react';
import ListContactContainer from './contact_container';

//redux objects
import configureStore from 'redux-mock-store'
import * as ContactActionTypes from '../../redux/actiontypes/actiontypes';
import * as ContactActions from '../../redux/actions/contacts_actions';
import * as ContactReducer from '../../redux/reducers/contacts_reducer';
import store from '../../redux/store/store';
const navigation = { navigate: jest.fn() };

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Contact List Container',()=>{
    it('renders correctly', () => {
        const tree = renderer.create(
          <ListContactContainer store={store} navigation={navigation}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
})