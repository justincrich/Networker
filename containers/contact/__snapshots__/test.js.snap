// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact List Container renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "position": "relative",
    }
  }
>
  <RCTScrollView
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    style={
      Object {
        "backgroundColor": undefined,
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "left": 0,
            "position": "relative",
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "display": "flex",
              "flexBasis": "auto",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
              "zIndex": 2000,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  false,
                  false,
                  false,
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  },
                  undefined,
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 40,
                  },
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "rgba(73, 88, 103, 1)",
                  "fontSize": 20,
                }
              }
            >
              New Contact
            </Text>
          </View>
          <Text
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            hitSlop={undefined}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "color": "rgba(73, 88, 103, 1)",
                "fontSize": 15,
                "fontWeight": "500",
              }
            }
            testID={undefined}
          >
            Save
          </Text>
        </View>
        <BVLinearGradient
          colors={
            Array [
              452984831,
              2164260863,
              4294967295,
            ]
          }
          end={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          locations={
            Array [
              0,
              0.1,
              1,
            ]
          }
          start={
            Object {
              "x": 0,
              "y": 1,
            }
          }
          style={
            Object {
              "flex": 1,
              "height": "100%",
              "position": "absolute",
              "width": "100%",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flex": 1,
            "flexDirection": "column",
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "marginBottom": 20,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(73, 88, 103, .5)",
                "borderRadius": 50,
                "borderWidth": 2,
                "height": 100,
                "justifyContent": "center",
                "overflow": "hidden",
                "position": "relative",
                "width": 100,
              }
            }
            testID={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                  "zIndex": 200,
                }
              }
            >
              <View
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(73, 88, 103, 1)",
                        "fontSize": 30,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        undefined,
                      ],
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "marginRight": 20,
            }
          }
        >
          <View
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 50,
                  "marginBottom": 5,
                }
              }
            >
              <View
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(73, 88, 103, .5)",
                        "fontSize": 40,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "width": 50,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "flexDirection": "column",
                  }
                }
              >
                <TextInput
                  onChangeText={[Function]}
                  placeholder="First Name"
                  style={
                    Object {
                      "borderBottomWidth": 1,
                      "borderColor": "rgba(73, 88, 103, .5)",
                      "fontSize": 16,
                      "height": 50,
                      "marginBottom": 10,
                      "overflow": "hidden",
                    }
                  }
                  underlineColorAndroid="rgba(0,0,0,0)"
                  value=""
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(253, 92, 99, 1)",
                        "fontSize": 12,
                        "position": "relative",
                        "top": -10,
                      },
                      Object {
                        "display": "none",
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 50,
                  "marginBottom": 5,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "width": 50,
                    },
                  ]
                }
              />
              <View
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "flexDirection": "column",
                  }
                }
              >
                <TextInput
                  onChangeText={[Function]}
                  placeholder="Last Name"
                  style={
                    Object {
                      "borderBottomWidth": 1,
                      "borderColor": "rgba(73, 88, 103, .5)",
                      "fontSize": 16,
                      "height": 50,
                      "marginBottom": 10,
                      "overflow": "hidden",
                    }
                  }
                  underlineColorAndroid="rgba(0,0,0,0)"
                  value=""
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(253, 92, 99, 1)",
                        "fontSize": 12,
                        "position": "relative",
                        "top": -10,
                      },
                      Object {
                        "display": "none",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 50,
                  "marginBottom": 5,
                }
              }
            >
              <View
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(73, 88, 103, .5)",
                        "fontSize": 40,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "width": 50,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "flexDirection": "column",
                  }
                }
              >
                <TextInput
                  onChangeText={[Function]}
                  placeholder="Company"
                  style={
                    Object {
                      "borderBottomWidth": 1,
                      "borderColor": "rgba(73, 88, 103, .5)",
                      "fontSize": 16,
                      "height": 50,
                      "marginBottom": 10,
                      "overflow": "hidden",
                    }
                  }
                  underlineColorAndroid="rgba(0,0,0,0)"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 50,
                  "marginBottom": 5,
                }
              }
            >
              <View
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(73, 88, 103, .5)",
                        "fontSize": 40,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "width": 50,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "flexDirection": "column",
                  }
                }
              >
                <TextInput
                  onChangeText={[Function]}
                  placeholder="Job Title"
                  style={
                    Object {
                      "borderBottomWidth": 1,
                      "borderColor": "rgba(73, 88, 103, .5)",
                      "fontSize": 16,
                      "height": 50,
                      "marginBottom": 10,
                      "overflow": "hidden",
                    }
                  }
                  underlineColorAndroid="rgba(0,0,0,0)"
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 50,
                  "marginBottom": 5,
                }
              }
            >
              <View
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(73, 88, 103, .5)",
                        "fontSize": 40,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "width": 50,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "flexDirection": "column",
                  }
                }
              >
                <TextInput
                  onChangeText={[Function]}
                  placeholder="Email"
                  style={
                    Object {
                      "borderBottomWidth": 1,
                      "borderColor": "rgba(73, 88, 103, .5)",
                      "fontSize": 16,
                      "height": 50,
                      "marginBottom": 10,
                      "overflow": "hidden",
                    }
                  }
                  underlineColorAndroid="rgba(0,0,0,0)"
                  value=""
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(253, 92, 99, 1)",
                        "fontSize": 12,
                        "position": "relative",
                        "top": -10,
                      },
                      Object {
                        "display": "none",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 50,
                  "marginBottom": 5,
                }
              }
            >
              <View
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(73, 88, 103, .5)",
                        "fontSize": 40,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "width": 50,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "flexDirection": "column",
                  }
                }
              >
                <TextInput
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Phone Number"
                  style={
                    Object {
                      "borderBottomWidth": 1,
                      "borderColor": "rgba(73, 88, 103, .5)",
                      "fontSize": 16,
                      "height": 50,
                      "marginBottom": 10,
                      "overflow": "hidden",
                    }
                  }
                  underlineColorAndroid="rgba(0,0,0,0)"
                  value=""
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(253, 92, 99, 1)",
                        "fontSize": 12,
                        "position": "relative",
                        "top": -10,
                      },
                      Object {
                        "display": "none",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 50,
                  "marginBottom": 5,
                }
              }
            >
              <View
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(73, 88, 103, .5)",
                        "fontSize": 40,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "width": 50,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "borderBottomWidth": 1,
                    "borderColor": "rgba(73, 88, 103, .5)",
                    "flex": 1,
                  }
                }
              >
                <TextInput
                  multiline={true}
                  numberOfLines={4}
                  onChangeText={[Function]}
                  placeholder="Notes"
                  style={undefined}
                  underlineColorAndroid="rgba(0,0,0,0)"
                  value=""
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
